app/api/groups/get_group_progress.go

```diff
@@ -72,67 +72,80 @@ func (srv *Service) getGroupProgress(w http.ResponseWriter, r *http.Request) ser
 		return service.NoError
 	}
 
-	var dbResult []map[string]interface{}
-	// It still takes about 3 minutes to complete on large data sets
-	service.MustNotBeError(srv.Store.GroupAncestors().
-		Select(`
-			groups_ancestors.idGroupAncestor AS idGroup,
+	users := srv.Store.GroupGroups().
+		Select("child.ID").
+		Joins("JOIN groups AS parent ON parent.ID = groups_groups.idGroupParent AND parent.sType != 'Team'").
```

Comment:
detail: I would find "clearer" (easier to read/understand) to have `parent.sType != 'Team'` and `child.sType = 'UserSelf'` and `groups_ancestors.idGroupAncestor IN (?)` moved to the "where" clause as they are more a filter than a joining rule. But don't change it if you prefer as it is now.